geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/geotools/GeoMesaFeatureSource.scala

```diff
@@ -59,7 +61,11 @@ class GeoMesaFeatureSource(val ds: DataStore with HasGeoMesaStats,
     import org.locationtech.geomesa.index.conf.QueryProperties.QueryExactCount
 
     val useExactCount = query.getHints.isExactCount.getOrElse(QueryExactCount.get.toBoolean)
-    val count = ds.stats.getCount(getSchema, query.getFilter, useExactCount).getOrElse(-1L)
+    val hints = new Hints()
+    query.getHints.getRequestedIndex.foreach { index => hints.put(QUERY_INDEX, index) }
+    hints.put(COST_EVALUATION, query.getHints.getCostEvaluation)
```

Comment:
I don't know if this should be passed through or not.

Reply:
seems like a good idea

Reply:
what do you think of having a Seq of hints to copy in the companion object? So something like:
```
val CountHints = Seq(QUERY_INDEX, COST_EVALUATION)
...
CountHints.foreach(key => if (query.getHints.contains(key)) { hints.put(key, query.getHints.get(key)) })
```
would make it easier to add/modify the hints we copy

Reply:
Sold!